src/utils/useAttachEvent.ts

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import * as stripeJs from '@stripe/stripe-js';
+
+export const useAttachEvent = <A extends unknown[]>(
+  elementRef: React.MutableRefObject<stripeJs.StripeElement | null>,
+  event: string,
+  cb?: (...args: A) => any
+) => {
+  React.useEffect(() => {
+    if (!cb || !elementRef.current) {
+      return () => {};
```

Comment:
I'm not sure if this needs to return anything - since whatever returned in a `useEffect` is just the cleanup function

Reply:
@jackieosborn-stripe  It feels like React is obliged to remember to fire a blank onUnmount.
I don't expect React to even fire empty function bodies, but just backing out seems --confusion && ++readability

```suggestion
      return;
```

BTW I dislike: `element as any` sorcery.
basically avoiding TS in a .ts file.

Reply:
The `return () => {};` was actually to get around an eslint issue:
`Arrow function expected no return value.eslint[consistent-return](https://eslint.org/docs/rules/consistent-return)`

@jackieosborn-stripe: Between `return;` and ignoring the error, vs `return () => {};`, are you ok to keep as-is? (Please lmk if there's a better way around this too!)

@faeb187: Thank you for your comment! Agreed on `(element as any)` not being ideal (or using `any` almost ever). However in this case, I believe it's a reasonable tradeoff over more elaborately checking the types here, especially since calling `element.on` with an unsupported event isn't breaking.